File: tests/fixtures/resend/resend-webhook-no-idempotency-broken/01-no-dedup.ts
import { Webhook } from 'svix';

const wh = new Webhook(process.env.RESEND_WEBHOOK_SECRET ?? '');

// Verifies the webhook but processes every delivery without deduplication.
export async function POST(req: Request) {
  const payload = await req.text();
  const headers = Object.fromEntries(req.headers as any);
  const event = wh.verify(payload, headers) as { type: string; data: { email_id: string } };

  if (event.type === 'email.delivered') {
    await recordDelivery(event.data);
  }

  return new Response('ok', { status: 200 });
}

async function recordDelivery(_data: unknown) {}

File: tests/fixtures/resend/resend-webhook-no-idempotency-broken/01-no-dedup.py
import os
from flask import Flask, request
from svix.webhooks import Webhook

app = Flask(__name__)
wh = Webhook(os.environ["RESEND_WEBHOOK_SECRET"])


@app.route("/webhook", methods=["POST"])
def handle_webhook():
    payload = request.get_data(as_text=True)
    event = wh.verify(
        payload,
        {
            "svix-id": request.headers.get("svix-id"),
            "svix-timestamp": request.headers.get("svix-timestamp"),
            "svix-signature": request.headers.get("svix-signature"),
        },
    )
    # Processes every delivery with no deduplication.
    if event.get("type") == "email.delivered":
        print("delivered")
    return {"ok": True}
